### PR TITLE
Refactor `OffsetDates`, `OffsetTimes`, `LocaDates` ,`LocalTimes`, `OffsetDateTime`.

### PR DESCRIPTION
**Pull request by alexander-aleksandrov**

### CODE REVIEW

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 1]
```diff
@@ -255,10 +253,41 @@ static LocalTime toLocalTime(Calendar calendar) {
         return LocalTimes.of(hours, minutes, seconds, millis);
     }
 
-    static Calendar toCalendar(LocalTime localTime) {
+    /**
+     * Obtains calendar using the data from offset date
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Calendars.java]:
```
Converts the passed {@code OffsetDate} into {@code Calendar}.

<p>The calendar is initialized at the offset from the passed date.
```

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 2]
```diff
@@ -255,10 +253,41 @@ static LocalTime toLocalTime(Calendar calendar) {
...
+     */
+    public static Calendar toCalendar(OffsetDate offsetDate) {
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Just `date`.

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 3]
```diff
@@ -255,10 +253,41 @@ static LocalTime toLocalTime(Calendar calendar) {
...
+        final Calendar calendar = at(offsetDate.getOffset());
+        setDate(calendar, offsetDate.getDate());
+        return calendar;
+    }
+
+    /**
+     * Obtains calendar using the data from local time
+     */
+    public static Calendar toCalendar(LocalTime localTime) {
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Just `time`.

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 4]
```diff
@@ -255,10 +253,41 @@ static LocalTime toLocalTime(Calendar calendar) {
...
         return createWithTime(localTime.getHours(),
                               localTime.getMinutes(),
                               localTime.getSeconds(),
                               localTime.getMillis());
     }
+
+    /**
+     * Obtains calendar using the data from local date
+     */
+    public static Calendar toCalendar(LocalDate localDate) {
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Just `date`.

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 5]
```diff
@@ -255,10 +253,41 @@ static LocalTime toLocalTime(Calendar calendar) {
...
+        return createDate(localDate.getYear(), localDate.getMonthValue(), localDate.getDay());
+    }
+
+    /**
+     * Obtains calendar using the data from offset time
+     */
+    public static Calendar toCalendar(OffsetTime offsetTime) {
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Just `time`.

File: values/src/main/java/org/spine3/time/Intervals.java [hunk 6]
```diff
@@ -35,7 +35,9 @@
  */
 public class Intervals {
 
-    private Intervals() {}
+    private Intervals() {
+
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/Intervals.java]: Why do you need the empty string here?

File: values/src/main/java/org/spine3/time/LocalTimes.java [hunk 7]
```diff
@@ -256,6 +256,14 @@ private static LocalTime changeMillis(LocalTime localTime, int millisDelta) {
         return add(localTime, MILLISECOND, millisDelta);
     }
 
+    /**
+     * Returns local time after changing one of the parameters on specified value.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalTimes.java]:
```
Performs time calculation using parameters of {@link Calendar#add()}. 
```

File: values/src/main/java/org/spine3/time/LocalTimes.java [hunk 8]
```diff
@@ -256,6 +256,14 @@ private static LocalTime changeMillis(LocalTime localTime, int millisDelta) {
...
+     *
+     * @param value local time that will be changed
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalTimes.java]: Remove params and return. It's not needed here.

File: values/src/main/java/org/spine3/time/OffsetDates.java [hunk 9]
```diff
@@ -153,28 +153,42 @@ private static OffsetDate changeMonth(OffsetDate offsetDate, int monthDelta) {
...
+     * @param daysDelta  a number of days that needs to be added or subtracted that can be either positive or negative
      * @return copy of this offset date with new days value
      */
     private static OffsetDate changeDays(OffsetDate offsetDate, int daysDelta) {
         final OffsetDate result = add(offsetDate, DAY_OF_MONTH, daysDelta);
         return result;
     }
 
+    /**
+     * Returns the same instance of offset date with changed local time parameter.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/OffsetDates.java]: It is not the same instance! It's a new one. Read documentation for `withXxx` methods from `java.time`.

File: values/src/main/java/org/spine3/time/OffsetDates.java [hunk 10]
```diff
@@ -153,28 +153,42 @@ private static OffsetDate changeMonth(OffsetDate offsetDate, int monthDelta) {
...
+    private static OffsetDate withDate(OffsetDate offsetDate, LocalDate localDate) {
+        return offsetDate.toBuilder()
+                         .setDate(localDate)
+                         .build();
+    }
+
+    /**
+     * Returns offset date changed on some value by specified parameter.
+     *
+     * @param offsetDate    instance of date that will be changed
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/OffsetDates.java]: Remove params and return docs. Not needed.

File: values/src/main/java/org/spine3/time/OffsetTimes.java [hunk 11]
```diff
@@ -226,18 +196,27 @@ private static OffsetTime changeSeconds(OffsetTime offsetTime, int secondsDelta)
...
-        final OffsetTime result = OffsetTime.newBuilder()
-                                            .setTime(localTime)
-                                            .setOffset(zoneOffset)
-                                            .build();
+        final OffsetTime result = add(offsetTime, MILLISECOND, millisDelta);
         return result;
     }
 
+    /**
+     * Returns offset time after changing one of the parameters on specified value.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/OffsetTimes.java]: `Performs time calculation using parameters of {@link Calendar#add()}.`

File: values/src/main/java/org/spine3/time/Intervals.java (hunk 6 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Intervals.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 12]
```diff
@@ -127,8 +127,15 @@ public static Calendar createDate(int year, int month, int day) {
         return calendar;
     }
 
+    /**
+     * Sets the date to calendar using local date.
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 13]
```diff
@@ -127,8 +127,15 @@ public static Calendar createDate(int year, int month, int day) {
...
+     *
+     * @param calendar this calendar will be set
+     * @param date     local date that should be set
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 14]
```diff
@@ -165,26 +172,10 @@ public static Calendar createWithTime(int hours, int minutes) {
     }
 
     /**
-     * Converts the passed {@code Calendar} to a new instance at the UTC time zone.
+     * Obtains a calendar with default GMT time zone.
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 15]
```diff
@@ -165,26 +172,10 @@ public static Calendar createWithTime(int hours, int minutes) {
...
+     *
+     * @return calendar instance
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 16]
```diff
@@ -201,13 +192,20 @@ public static Calendar nowAt(ZoneOffset zoneOffset) {
         final Calendar calendar = at(zoneOffset);
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java [hunk 17]
```diff
@@ -201,13 +192,20 @@ public static Calendar nowAt(ZoneOffset zoneOffset) {
...
+        int offsetFromGmt = calendar.getTimeZone()
+                                    .getOffset(msFromEpoch);
         calendar.setTime(timeAtGmt);
         calendar.add(MILLISECOND, offsetFromGmt);
 
         return calendar;
     }
 
+    /**
+     * Obtains calendar at specified zone offset
```
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java (hunk 1 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java (hunk 2 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java (hunk 3 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java (hunk 4 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/Calendars.java (hunk 5 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/Calendars.java]: Fixed

File: values/src/main/java/org/spine3/time/LocalTimes.java (hunk 8 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/LocalTimes.java]: Fixed

File: values/src/main/java/org/spine3/time/LocalTimes.java (hunk 7 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/LocalTimes.java]: Fixed

File: values/src/main/java/org/spine3/time/OffsetDates.java (hunk 9 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/OffsetDates.java]: Fixed

File: values/src/main/java/org/spine3/time/OffsetDates.java (hunk 10 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/OffsetDates.java]: Fixed

File: values/src/main/java/org/spine3/time/OffsetTimes.java (hunk 11 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/OffsetTimes.java]: Fixed

**Pull request merged**